src/Data/Project.hs

```diff
@@ -35,16 +35,26 @@ projectExtensions = extensionsForLanguage . projectLanguage
 projectFiles :: Project -> [File]
 projectFiles = fmap blobFile . projectBlobs
 
-readProjectFromPaths :: MonadIO m => Maybe FilePath -> FilePath -> Language -> [FilePath] -> m Project
+readProjectFromPaths :: MonadIO m
+                     => Maybe Path.AbsRelDir -- ^ An optional root directory for the project
+                     -> Path.AbsRelFileDir   -- ^ A file or directory to parse. Passing a file path loads all files in that file's parent directory.
```

Comment:
I gotta say, I’m not sold on this semantic, but it looks to me like you’re maintaining the current behaviour and not introducing this so let’s see about resolving it separately from this PR.

Reply:
Yeah, it’s pretty shifty! Revealing weirdnesses like this is one of my favorite parts of adding `pathtype` everywhere: it lays bare the sins were committing and not even knowing it.